lib/swagger-explorer.ts

```diff
@@ -353,6 +353,8 @@ export class SwaggerExplorer {
       ...omit(requestBody, keysToRemove),
       ...this.mimetypeContentWrapper.wrap(consumes, pick(requestBody, 'schema'))
     };
+    if (requestBody.name)
+      document.root["x-codegen-request-body-name"] = requestBody.name;
```

Comment:
Why is this needed?

Reply:
If the property is not set, the default value will be used (the class name). see https://openapi-generator.tech/docs/swagger-codegen-migration/#body-parameter-name. With this change and the `@ApiBody({name: "..."})` which we still have in our code from @nestjs/swagger@3.x, our generated APIs look exactly like before.

Reply:
Can I provide any additional information @kamilmysliwiec to support you in your merge decision? 

I know that the generators from openapi-generator.tech aren't the only api generators out there and `x-codegen-request-body-name` seems to be a non-standard for now. But it seems that others are picking up that property as well, see https://github.com/swagger-api/swagger-codegen-generators/pull/827/files